From 2866d7c2509d1d83c749231c9d60b174cfa7d841 Mon Sep 17 00:00:00 2001
From: Vyacheslav Bocharov <adeep@lexina.in>
Date: Wed, 31 May 2023 14:35:47 +0300
Subject: [PATCH 03/12] enable DT overlays support

---
 arch/arm/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 9898c7d68e..2ee416844e 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -850,6 +850,8 @@ config ARCH_OMAP2PLUS
 config ARCH_MESON
 	bool "Amlogic Meson"
 	select GPIO_EXTRA_HEADER
+	select OF_LIBFDT
+	select OF_LIBFDT_OVERLAY
 	imply DISTRO_DEFAULTS
 	imply DM_RNG
 	help
-- 
2.34.1

